src/main/java/ch/hearc/todont/models/Pledge.java

```diff
@@ -47,6 +48,20 @@
 
     // METHODS
 
+    /**
+     * Whether the ToDont has been done or not
+     * 
+     * @return True if the user has failed the ToDont
+     */
+    public boolean isFailed() {
+        if (dateFailed == null) {
+            return false;
+        }
+        return Timestamp.from(Instant.now()).after(dateFailed);
```

Comment:
If it has a fail date, can't we assume it's failed? As far as I know we don't have any plan for the user to be able to plan a failing date?

Reply:
I thought so as well. I decided to write the test anyway because of readability and semantics. Should I remove it ?

Reply:
I would probably just replace it by `return dataFailed != null;`, but it could also make sense to keep the way it is now, so do as you wish (or we could wait for another point of view)

Reply:
Sure